src/python_testing/TC_MCORE_FS_1_4.py

```diff
@@ -48,7 +48,34 @@
 from mobly import asserts
 
 
+async def wait_for_server_initialization(server_port, timeout=5):
+    """Wait until the server is ready by checking if it opens the expected port."""
+    start_time = asyncio.get_event_loop().time()
+    elapsed_time = 0
+    retry_interval = 1
+
+    logging.info(f"Waiting for server to initialize on TCP port {server_port} for up to {timeout} seconds.")
```

Comment:
What guarantees that TCP is even supported by that server?

Reply:
We use dedicated test server "lighting-app-data-mode-no-unique-id"

Reply:
Yes, but where is it documented/guaranteed that it supports TCP?  That is, what keeps this from randomly breaking?

Reply:
The TCP is enabled by default. https://github.com/project-chip/connectedhomeip/blob/master/src/inet/inet.gni#L27
As long as we don't explicitly override this value in our example app, then TCP is enabled, Non-Linux platforms disable it currently, but they do it explicitly.

Reply:
@yufengwangca Yes, but:

1. TH gets run on non-linux platforms too.
2. Absolutely nothing ensures that Linux will keep enabling TCP by default.

Reply:
In this case, we're checking the dedicated example app "lighting-app-data-mode-no-unique-id," which only has a Linux implementation.

I can switch to checking the specific log output from stdout instead, but I’m not convinced this is a better approach. It would be even more fragile if the log format changes.

Pradip mentioned that TCP is enabled by default, and we need to explicitly override this flag. Therefore, we can add a comment in the example app to avoid overriding it, as the example app is completely under our control.

Reply:
> In this case, we're checking the dedicated example app "lighting-app-data-mode-no-unique-id," which only has a Linux implementation.

The "linux" bits are built on both Linux and Mac.  They should really be called "posix" or something, not "linux", but here we are.

Reply:
https://github.com/project-chip/connectedhomeip/pull/35547